stdpopsim/slim_engine.py

```diff
@@ -1249,6 +1230,15 @@ def _recap_and_rescale(
             )
             for pop in recap_epoch.populations
         ]
+        #########################################
+        # FIX THIS BEFORE MERGING INTO MASTER!!!!
+        # NEED THE LATEST SLiM
+        #########################################
+        # TODO: remove this line once SLiM starts populating mutation times
+        tables = ts.tables
+        tables.compute_mutation_times()
+        ts = pyslim.load_tables(tables)
+        # TODO: using recombination map in recapitate. maps should be discrete.
         ts = ts.recapitate(
```

Comment:
is this working with all the updates in pyslim/msprime1.0?

Reply:
Umm... I don't know. Maybe just try it in and see?

Reply:
see https://github.com/petrelharp/pyslim/blob/new_recapitate/pyslim/slim_tree_sequence.py#L391
- figure out slim generation
self.metadata['SLiM']['generation']
- simulate before that with sim_ancestry
    - set populations configs
   msprime.Demography.from_tree_sequence(ts)
- add mutations
-  see if we are forcing there to be a neutral mut type
- create a slim mutation model with that type, REMEMBER to set the next mutation id (it's in the derived state column)
max_id = -1
for mut in ts.mutations():
  for d in mut.derived_state.split(","):
     max_id = max(max_id, int(d))

maybe there is a way to unpack all the derived state columns at once and then used vectorized opretations (np)